src/main/java/com/zaxxer/nuprocess/windows/WindowsProcess.java

```diff
@@ -270,6 +246,64 @@ NuProcess start(List<String> commands, String[] environment, Path cwd)
       return this;
    }
 
+   void run(List<String> commands, String[] environment, Path cwd)
+   {
+      callPreStart();
+
+      try {
+         prepareProcess(commands, environment, cwd);
+
+         myProcessor = new ProcessCompletions(this);
+
+         callStart();
+
+         NuKernel32.ResumeThread(processInfo.hThread);
+      }
+      catch (Throwable e) {
+         e.printStackTrace();
+         onExit(Integer.MIN_VALUE);
+         return;
+      }
+      finally {
+         NuKernel32.CloseHandle(hStdinWidow);
+         NuKernel32.CloseHandle(hStdoutWidow);
+         NuKernel32.CloseHandle(hStderrWidow);
+      }
+
+      // This must happen after the widow handles are closed; otherwise the process loop will
+      // never end because stderr and stdout will never be detected as closed
+      myProcessor.run();
+   }
+
+   private void prepareProcess(List<String> commands, String[] environment, Path cwd)
```

Comment:
This is the same logic that was previously in `WindowsProcess.start`, refactored into a separate method for reuse in `run` but otherwise unchanged